hikari/impl/buckets.py

```diff
@@ -303,6 +333,24 @@ def drip(self) -> None:
         if not self.is_unknown:
             self.remaining -= 1
 
+    def resolve(self, real_bucket_hash: str) -> None:
+        """Resolve an unknown bucket.
+
+        Parameters
+        ----------
+        real_bucket_hash: builtins.str
+            The real bucket hash for this bucket.
+
+        Raises
+        ------
+        builtins.RuntimeError
+            If the hash of the bucket is already known.
+        """
+        if not self.is_unknown:
+            raise RuntimeError("Cannot resolve known bucket")
```

Comment:
Is there any point in resolving them if it doesn't ensure that ratelimitimg for a single bucket will be handled by and locked the main bucket object now that we’ve discovered the overlap

Reply:
When we make a request to an unknown bucket. This will handle that bucket under an unknown hash. Every other request to that same route will get stopped until the lock is released, after which we will (if nothing went wrong) know what the ratelimits are. We will then resolve the bucket and all new requests and existing requests waiting for the lock will go under that same bucket, respecting the ratelimits.

Reply:
And if multiple requests are locked to a unknown bucket and it then gets resolved to an already existing bucket youll now have two active request queues for the one bucket

Reply:
It should not resolve to an already existing bucket 🤔(unless im missing something)

Reply:
Multiple endpoints can have the same bucket hash

Reply:
Oh, ig I didn't account for those. Will fix

Reply:
This behaviour was being handled in the ver I made of this so could just use that as a reference